src/ledger-bridge-keyring.ts

```diff
@@ -0,0 +1,908 @@
+        : true;
+      accounts.push({
+        address,
+        balance: null,
+        index: i,
+      });
+
+      // PER BIP44
+      // "Software should prevent a creation of an account if
+      // a previous account does not have a transaction history
+      // (meaning none of its addresses have been used before)."
+      if (!valid) {
+        break;
+      }
+    }
+    return accounts;
+  }
+
+  _getAccountsLegacy(from: number, to: number) {
+    const accounts: {
+      address: string;
+      balance: number | null;
+      index: number;
+    }[] = [];
+
+    for (let i = from; i < to; i++) {
+      const address = this._addressFromIndex(pathBase, i);
+      accounts.push({
+        address,
+        balance: null,
+        index: i,
+      });
+      this.paths[ethUtil.toChecksumAddress(address)] = i;
+    }
+    return accounts;
+  }
+
+  _padLeftEven(hex: string) {
+    return hex.length % 2 === 0 ? hex : `0${hex}`;
+  }
+
+  _normalize(buf: Buffer) {
+    return this._padLeftEven(ethUtil.bufferToHex(buf).toLowerCase());
+  }
+
+  // eslint-disable-next-line no-shadow
+  _addressFromIndex(basePath: string, i: number) {
+    const dkey = this.hdk.derive(`${basePath}/${i}`);
+    const address = ethUtil
+      .publicToAddress(dkey.publicKey, true)
+      .toString('hex');
+    return ethUtil.toChecksumAddress(`0x${address}`);
+  }
+
+  _pathFromAddress(address: string) {
+    const checksummedAddress = ethUtil.toChecksumAddress(address);
+    let index = this.paths[checksummedAddress];
+    if (typeof index === 'undefined') {
+      for (let i = 0; i < MAX_INDEX; i++) {
+        if (checksummedAddress === this._addressFromIndex(pathBase, i)) {
+          index = i;
+          break;
+        }
+      }
+    }
+
+    if (typeof index === 'undefined') {
+      throw new Error('Unknown address');
+    }
+    return this._getPathForIndex(index);
+  }
+
+  _toAscii(hex: string) {
+    let str = '';
+    let i = 0;
+    const l = hex.length;
+    if (hex.substring(0, 2) === '0x') {
+      i = 2;
+    }
+
+    for (; i < l; i += 2) {
+      const code = parseInt(hex.substr(i, 2), 16);
+      str += String.fromCharCode(code);
+    }
+
+    return str;
+  }
+
+  _getPathForIndex(index: number) {
+    // Check if the path is BIP 44 (Ledger Live)
+    return this._isLedgerLiveHdPath()
+      ? `m/44'/60'/${index}'/0/0`
+      : `${this.hdPath}/${index}`;
+  }
+
+  _isLedgerLiveHdPath() {
+    return this.hdPath === `m/44'/60'/0'/0/0`;
+  }
+
+  _toLedgerPath(path: string) {
+    return path.toString().replace('m/', '');
+  }
+
+  async _hasPreviousTransactions(address: string) {
+    const apiUrl = this._getApiUrl();
+    const response = await window.fetch(
+      `${apiUrl}/api?module=account&action=txlist&address=${address}&tag=latest&page=1&offset=1`,
+    );
+    const parsedResponse = await response.json();
+    if (parsedResponse.status !== '0' && parsedResponse.result.length > 0) {
```

Comment:
Nit: Again, maybe out of the scope of the MR, but this `if` statement is a bit verbose. It could be safely refactored to something that, IMO, is more readable, like this:
```ts
    const parsedRes = await response.json();
    const resExists = parsedRes.status !== '0' && parsedRes.result?.length;

    return resExists;
```

At the very least, it could just be shortened to

```ts
return parsedResponse.status !== '0' && parsedResponse.result.length > 0
```